File: Data Structures & Algorithms/longest-palindromic-substring/submission-4.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        if(s.size() == 1) return s;
        string res = "";
        for(int i=1; i<s.size(); i++){
            int l = i;
            int r = i;
            while(l>-1 && r<s.size() && s[l] == s[r]){
                l--;
                r++;
            }
            string p = s.substr(l+1, r-l-1);
            if(p.size() > res.size()){
                res = p;
            }
            l = i-1;
            r = i;
            while(l>-1 && r<s.size() && s[l] == s[r]){
                l--;
                r++;
            }
            p = s.substr(l+1, r-l-1);
            if(p.size() > res.size()){
                res = p;
            }
            
        }

        return res;

    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        i,j=0,0
        mp = defaultdict(int)
        res = 0
        while(j<len(s)):
            mp[s[j]] += 1
            ch = max(mp, key=mp.get)
            print(ch)
            if ((j-i+1)-mp[ch]) <= k:
                res = max(res, j-i+1)
                j+=1
            else:
                while ((j-i+1)-mp[ch]) > k:
                    mp[s[i]]-=1
                    if mp[s[i]] == 0:
                        del mp[s[i]]
                    i+=1
                    ch = max(mp, key=mp.get)
                j+=1
        return res

            


